src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -2137,7 +2137,7 @@ uct_ib_mlx5_devx_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
         return UCS_ERR_INVALID_PARAM;
     }
 
-    if (flags & UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA) {
+    if ((flags & UCT_MD_MKEY_PACK_FLAG_INVALIDATE_RMA) || (memh->dm != NULL)) {
```

Comment:
Is this done intentionally? If so, why is this needed?

Reply:
Device memory depends on an indirect key, the only way to access it is through RDMA operations.